mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,492 @@
+# Authors: Mark Wronkiewicz <wronk.mark@gmail.com>
+#          Jussi Nurminen <jnu@iki.fi>
+
+# This code was adapted and relicensed (with BSD form) with permission from
+# Jussi Nurminen
+
+# License: BSD (3-clause)
+
+# Note, there are an absurd number of different possible notations for
+# spherical coordinates, which confounds the notation for spherical harmonics.
+# Here, we purposefully stay away from shorthand notation in both and use
+# explicit terms (like 'azimuth' and 'polar') to avoid confusion.
+
+# TODO: write in equation numbers from Samu's paper
+
+from __future__ import division
+import numpy as np
+from scipy.special import lpmv
+from scipy.linalg import pinv
+from scipy.misc import factorial as fact
+from ..utils import logger
+from .. import pick_types, pick_info
+from ..io.constants import FIFF
+from ..forward._compute_forward import _concatenate_coils
+from ..forward._make_forward import _read_coil_defs, _create_coils
+
+
+def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3, n_jobs=1):
+    """Apply Maxwell filter to data using spherical harmonics.
+
+    Parameters
+    ----------
+    raw : instance of mne.io.Raw
+        Data to be filtered
+    origin : list, tuple, or ndarray, shape (3,)
+        Origin of internal and external multipolar moment space in millimeters
+    int_order : int
+        Order of internal component of spherical expansion
+    ext_order : int
+        Order of external component of spherical expansion
+    n_jobs : int
+        Number of jobs to run in parallel
+
+    Returns
+    -------
+    raw_sss : instance of mne.io.Raw
+        The raw data with Maxwell filtering applied
+    """
+
+    # TODO: Add error checks on input parameters
+
+    # Create coil information
+    all_coils, meg_info = _make_coils(raw.info, accurate=True)
+    picks = [raw.info['ch_names'].index(ch) for ch in [coil['chname']
+                                                       for coil in all_coils]]
```

Comment:
should be simpler just to do `[raw.info['ch_names'].index[coil['chname']] for coil in all_coils]`